app/src/main/java/ch/epfl/qedit/viewmodel/QuizViewModel.java

```diff
@@ -0,0 +1,56 @@
+package ch.epfl.qedit.viewmodel;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+import ch.epfl.qedit.backend.database.DatabaseFactory;
+import ch.epfl.qedit.model.Quiz;
+import ch.epfl.qedit.util.BundledData;
+import ch.epfl.qedit.util.Callback;
+import ch.epfl.qedit.util.Response;
+
+public class QuizViewModel extends ViewModel {
+    public enum Status {
+        NotLoaded,
+        Loading,
+        Loaded,
+        CouldntLoad
```

Comment:
But here, you create the enum... ?

Reply:
That enum is not related to a service (auth or database) but rather to the state of the quiz activity. It is used by fragments to react when the data is loading from the database (progress bar), and to print a toast in case of error (couldn't load)